pcsx2/ImGui/FullscreenUI.cpp

```diff
@@ -4592,6 +5045,520 @@ void FullscreenUI::DrawMemoryCardSettingsPage()
 		}
 	}
 
+	EndMenuButtons();
+}
+
+void FullscreenUI::DrawNetworkHDDSettingsPage()
+{
+	static constexpr const char* eth_api_names[] = {
+		FSUI_NSTR("Unset"),
+		FSUI_NSTR("PCAP Bridged"),
+		FSUI_NSTR("PCAP Switched"),
+		FSUI_NSTR("TAP"),
+		FSUI_NSTR("Sockets"),
+	};
+
+	static constexpr const char* eth_api_values[] = {
+		"Unset",
+		"PCAP Bridged",
+		"PCAP Switched",
+		"TAP",
+		"Sockets",
+	};
+
+	static constexpr const char* dns_options[] = {
+		FSUI_NSTR("Manual"),
+		FSUI_NSTR("Auto"),
+		FSUI_NSTR("Internal"),
+	};
+
+	static constexpr const char* dns_values[] = {
+		"Manual",
+		"Auto",
+		"Internal",
+	};
+
+	SettingsInterface* bsi = GetEditingSettingsInterface();
+
+	BeginMenuButtons();
+
+	MenuHeading(FSUI_CSTR("Network Adapter"));
+
+	DrawToggleSetting(bsi, FSUI_ICONSTR(ICON_FA_NETWORK_WIRED, "Enable Network Adapter"),
+		FSUI_CSTR("Enables the network adapter for online functionality and LAN play."), "DEV9/Eth", "EthEnable", false);
+
+	const bool network_enabled = GetEffectiveBoolSetting(bsi, "DEV9/Eth", "EthEnable", false);
+
+	// Get current API to determine which adapters to show
+	const std::string current_api = bsi->GetStringValue("DEV9/Eth", "EthApi", "Unset");
+	
+	static std::vector<std::vector<AdapterEntry>> adapter_lists;
+	static std::vector<Pcsx2Config::DEV9Options::NetApi> api_types;
+	static std::vector<std::string> api_display_names;
+	static bool adapters_loaded = false;
+	
+	if (!adapters_loaded && network_enabled) // Load adapters when network gets enabled
+	{
+		adapter_lists.clear();
+		api_types.clear();
+		api_display_names.clear();
+		
+		// Unset (empty list)
+		adapter_lists.emplace_back();
+		api_types.emplace_back(Pcsx2Config::DEV9Options::NetApi::Unset);
+		api_display_names.emplace_back("Unset");
+		
+		// PCAP adapters
+		std::vector<AdapterEntry> pcap_adapters = PCAPAdapter::GetAdapters();
+		if (!pcap_adapters.empty())
+		{
+			std::vector<AdapterEntry> unique_pcap_adapters;
+			std::set<std::string> seen_guids;
+			
+			for (const auto& adapter : pcap_adapters)
+			{
+				if (seen_guids.find(adapter.guid) == seen_guids.end())
+				{
+					seen_guids.insert(adapter.guid);
+					unique_pcap_adapters.push_back(adapter);
+				}
+			}
+			
+			if (!unique_pcap_adapters.empty())
+			{
+				adapter_lists.emplace_back(unique_pcap_adapters);
+				api_types.emplace_back(Pcsx2Config::DEV9Options::NetApi::PCAP_Bridged);
+				api_display_names.emplace_back("PCAP Bridged");
+				
+				adapter_lists.emplace_back(unique_pcap_adapters);
+				api_types.emplace_back(Pcsx2Config::DEV9Options::NetApi::PCAP_Switched);
+				api_display_names.emplace_back("PCAP Switched");
+			}
+		}
```

Comment:
Technically speaking
you need to sort the returned vector into a vector for switched and a vector for bridged
As they are currently a 1-1 duplicate, you get the right answer even with the wrong working out

Reply:
Sounds good